contrib/guix/libexec/build.sh

```diff
@@ -111,7 +134,8 @@ make -C depends --jobs="$MAX_JOBS" HOST="$HOST" \
                                    x86_64_linux_RANLIB=x86_64-linux-gnu-ranlib \
                                    x86_64_linux_NM=x86_64-linux-gnu-nm \
                                    x86_64_linux_STRIP=x86_64-linux-gnu-strip \
-                                   qt_config_opts_i686_linux='-platform linux-g++ -xplatform bitcoin-linux-g++'
+                                   qt_config_opts_i686_linux='-platform linux-g++ -xplatform bitcoin-linux-g++' \
+                                   mingw32_CFLAGS='-DMINGW_HAS_SECURE_API=1 -pipe'
```

Comment:
Should we also be setting `MINGW_HAS_SECURE_API` for regular builds? I remember you talking about this potentially fixing an issue you were seeing. Can you remember what that was?

Reply:
Pre-https://github.com/mirror/mingw-w64/commit/3bef7c2206bb6f9552ea7e61315c4bf7af3aa6c9 (first included in mingw-w64 v7.0.0), we had to either `-DMINGW_HAS_SECURE_API=1` or specify `--enable-secure-api` at configure time. Debian [does the latter](https://salsa.debian.org/mingw-w64-team/mingw-w64/-/blob/debian/6.0.0-4/debian/rules#L101).

So I think I'll bump to mingw-w64 v7.0.0 in this PR, I was initially hesitant as it seems v7.0.0 won't be as battle-tested as v6.0.0, but it seems Ubuntu 20.04 LTS [will be releasing with v7.0.0](https://packages.ubuntu.com/source/focal/mingw-w64), so perhaps that's a problem that will solve itself. Plus we'll get the `_FORTIFY_SOURCE` support.

Reply:
Context, waiting to rebase over #18358 once we figure out the best solution there. I've tested with cherry-picking existing solutions in that PR and the build seems to work.